src/allmydata/web/status.py

```diff
@@ -1128,49 +1129,92 @@ def childFactory(self, ctx, name):
                 if s.get_counter() == count:
                     return RetrieveStatusPage(s)
 
-
-class HelperStatus(MultiFormatPage):
-    docFactory = getxmlfile("helper.xhtml")
+# Render "/helper_status" page.
+class HelperStatus(MultiFormatResource):
 
     def __init__(self, helper):
-        rend.Page.__init__(self, helper)
-        self.helper = helper
+        super(HelperStatus, self).__init__()
+        self._helper = helper
 
-    def data_helper_stats(self, ctx, data):
-        return self.helper.get_stats()
+    def render_HTML(self, req):
+        return renderElement(req, HelperStatusElement(self._helper))
 
     def render_JSON(self, req):
         req.setHeader("content-type", "text/plain")
-        if self.helper:
-            stats = self.helper.get_stats()
+        if self._helper:
+            stats = self._helper.get_stats()
             return json.dumps(stats, indent=1) + "\n"
         return json.dumps({}) + "\n"
 
-    def render_active_uploads(self, ctx, data):
-        return data["chk_upload_helper.active_uploads"]
+class HelperStatusElement(Element):
 
-    def render_incoming(self, ctx, data):
-        return "%d bytes in %d files" % (data["chk_upload_helper.incoming_size"],
-                                         data["chk_upload_helper.incoming_count"])
+    loader = XMLFile(FilePath(__file__).sibling("helper.xhtml"))
 
-    def render_encoding(self, ctx, data):
-        return "%d bytes in %d files" % (data["chk_upload_helper.encoding_size"],
-                                         data["chk_upload_helper.encoding_count"])
+    def __init__(self, helper):
+        """
+        :param _allmydata.immutable.offloaded.Helper helper
+        """
+        super(HelperStatusElement, self).__init__()
+        self._helper = helper
+
```

Comment:
It might make this whole class a little more clear if it just passed in `data` (that is, the stuff we actually need) instead of `helper` (that is, a thing that can tell us what we actually want).

Reply:
I had tried this:
```python
    def __init__(self, helper):
        super(HelperStatusElement, self).__init__()
        self._data = helper.get_stats()
```
That crashes when `helper` is `None`, when helper isn't running.  This way we get to report that the helper isn't running, from `helper_running()` renderer.

Reply:
My idea was something like this:

```
    def __init__(self, data):
        super(HelperStatusElement, self).__init__()
        self._data = data

    def helper_running(self):
        if self._data is None:
            return "not running"

# ...
status = HelperStatus(helper.get_stats() if helper else None)
```
...but also that's more "clean-up" than "porting" so if that doesn't work, that's cool too :)